src/theme-provider.js

```diff
@@ -53,13 +58,12 @@ class ThemeProvider extends React.Component {
     // set broadcast state by merging outer theme with own
     if (this.context[CHANNEL]) {
       this.setOuterTheme(this.context[CHANNEL].getState())
-      this.broadcast.setState(this.getTheme())
```

Comment:
Can you explain why this was removed? It seems unrelated.

Reply:
publishTheme method is doing a setState and setOuterTheme uses now publishTheme under the hood

Reply:
I think this is the real fix, and `publishTheme` is only incidentally changed.

This line was setting the state with no argument, which defaults to the `props.theme` here https://github.com/paypal/glamorous/blob/cb6de3abe6b3a7e2643828f4239e0487d4f80455/src/theme-provider.js#L21, meaning it's not setting the theme to the *next* theme (it's just resetting it to the theme it already has).

A lot of the theming work was done by @vesparny, so if I'm wrong hopefully he can correct me. 😄

Reply:
Yeah, this part of defaulting to `this.props.theme` is a making a control flow a little bit iffy but I left this part untouched. Personally I would prefer explicit arguments, so that `publishTheme` would always have to receive `getTheme`'s result as argument which in turn would always be called with `this.props.theme` if necessary.

Also I've noticed a different small issue too - broadcast is initialized as:
```js
broadcast = brcast(this.props.theme)
```
but that gives it untrue initialState in case of nested ThemeProvider. I think creation of the `broadcast` should be delayed.

Reply:
> but that gives it untrue initialState in case of nested ThemeProvider. I think creation of the broadcast should be delayed.

I'm pretty sure I follow, but to be clear, are you suggesting moving it to something like `componentWillMount`? With this PR, the theme that was (possibly) merged with the outer theme is now going to be set properly inside `componentWillMount()`, so that untrue initial state won't really ever be in effect.

Did I follow your point correctly? If not any clarification would be great. 🙌

Reply:
Yeah, that was exactly my point :)

Reply:
Cool, thanks! It makes sense to me. Maybe you want to do that as a follow-up PR?

Reply:
Sure thing, would prefer to create it once we settle on this one to avoid potential conflicts

Reply:
Thanks for the discussion here!